internal/apiserver/pgadminservice/pgadminimpl.go

```diff
@@ -61,14 +61,62 @@ func CreatePgAdmin(request *msgs.CreatePgAdminRequest, ns, pgouser string) msgs.
 			return resp
 		}
 
+		// set the default pgadmin storage value (i.e. the primary storage value)
+		if cluster.Spec.PGAdminStorage == (crv1.PgStorageSpec{}) {
+			cluster.Spec.PGAdminStorage = cluster.Spec.PrimaryStorage
+			log.Info("pgAdmin storage not set, using primary storage configuration")
+		}
+
+		// if a value for pgAdmin storage config is provided with the request, validate it here
+		// if it is not valid, return now
+		if request.StorageConfig != "" && !apiserver.IsValidStorageName(request.StorageConfig) {
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = fmt.Sprintf("%q storage config provided invalid", request.StorageConfig)
+			return resp
+		}
+
+		// validate the pgAdmin storage config value in the main configuration if the request did not provide one
+		// if it is set but not valid, return now
+		if request.StorageConfig == "" && apiserver.Pgo.PGAdminStorage != "" &&
+			!apiserver.IsValidStorageName(apiserver.Pgo.PGAdminStorage) {
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = fmt.Sprintf("%q configured storage config invalid", apiserver.Pgo.PGAdminStorage)
+			return resp
+		}
+
+		// if the PVCSize is set to a customized value, ensure that it is recognizable by Kubernetes
+		if err := apiserver.ValidateQuantity(request.PVCSize); request.PVCSize != "" && err != nil {
+			resp.Status.Code = msgs.Error
+			resp.Status.Msg = fmt.Sprintf(apiserver.ErrMessagePVCSize, request.PVCSize, err.Error())
+			return resp
+		}
```

Comment:
I would move this into the `if request.PVCSize != ""` block to keep it all together, and to ensure that we've set the StorageConfig at that point.

Reply:
makes sense to me, updated.